### DISCUSSION TITLE
Geometry.save is not an app

### DISCUSSION THREAD

**Discussion by tbraeckevelt** [molmod/psiflow#37] [Q&A] [answered]
I was wondering why Geometry.save() is not a python_app?

I wrote a wrapper around the function so that I could save future Geometries:

```
@typeguard.typechecked
def _save_futgeo(geo: Geometry, path_out: Union[Path, str]):
    geo.save(path_out)

save_futgeo = python_app(_save_futgeo, executors=["default_threads"])
```

But this might not be necessary if  Geometry.save()  is already a python_app.

**Comment by svandenhaute** [comment, marked as answer]
`save()` is an instance method, so you need to call it on an existing geometry, you cannot call it on a future of a geometry:

```python
geometry = ...  # 
geometry.save(my_path)  # OK
future_of_geometry.save(my_path)  # will not work because Parsl Futures do not know what `save` is
```

To avoid writing an app for this, you can use the `Dataset` functionality, which works in both cases:

```python
Dataset([future_of_geometry]).save(path_out)
```
